### PR TITLE
Check error message for raised exception

### PR DESCRIPTION
**Pull request by sumanau7**
- [x] xref #30999
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by sumanau7** [comment]
@MomIsBestFriend Please review.

File: pandas/tests/indexes/period/test_setops.py [hunk 1]
```diff
@@ -148,7 +148,8 @@ def test_union_misc(self, sort):
         # raise if different frequencies
         index = period_range("1/1/2000", "1/20/2000", freq="D")
         index2 = period_range("1/1/2000", "1/20/2000", freq="W-WED")
-        with pytest.raises(IncompatibleFrequency):
+        msg = fr".*Input has different freq=W-WED from PeriodIndex/(freq=D/)"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_setops.py]: does the ```.*``` necessary here?

Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_setops.py]:
```suggestion
        msg = fr".*Input has different freq=W-WED from PeriodIndex\(freq=D\)"
```

Comment (sumanau7) [reply pandas/tests/indexes/period/test_setops.py]: No, updated the PR.

**Comment by sumanau7** [comment]
@MomIsBestFriend Green.

**Comment by MomIsBestFriend** [comment]
>     * [x]  closes #30999
> 
>     * [x]  tests added / passed
> 
>     * [x]  passes `black pandas`
> 
>     * [x]  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
> 
>     * [ ]  whatsnew entry

@sumanau7 Can you please change the word "closes" to the word "xref"?

File: pandas/core/arrays/period.py [hunk 2]
```diff
@@ -166,7 +166,8 @@ def __init__(self, values, freq=None, dtype=None, copy=False):
     @classmethod
     def _simple_new(cls, values: np.ndarray, freq=None, **kwargs) -> "PeriodArray":
         # alias for PeriodArray.__init__
-        assert isinstance(values, np.ndarray) and values.dtype == "i8"
+        assertion_msg = f"Should be numpy array of type i8"
```
Comment (MomIsBestFriend) [inline pandas/core/arrays/period.py]:
```suggestion
        assertion_msg = "Should be numpy array of type i8"
```

File: pandas/tests/indexes/period/test_constructors.py [hunk 3]
```diff
@@ -327,7 +327,8 @@ def test_constructor_simple_new(self):
         result = idx._simple_new(idx._data, name="p")
         tm.assert_index_equal(result, idx)
 
-        with pytest.raises(AssertionError):
+        msg = f"Should be numpy array of type i8"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_constructors.py]:
```suggestion
        msg = "Should be numpy array of type i8"
```

File: pandas/tests/indexes/period/test_join.py [hunk 4]
```diff
@@ -39,5 +39,6 @@ def test_join_does_not_recur(self):
     def test_join_mismatched_freq_raises(self):
         index = period_range("1/1/2000", "1/20/2000", freq="D")
         index3 = period_range("1/1/2000", "1/20/2000", freq="2D")
-        with pytest.raises(IncompatibleFrequency):
+        msg = fr".*Input has different freq=2D from PeriodIndex\(freq=D\)"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_join.py]:
```suggestion
        msg = r"Input has different freq=2D from PeriodIndex\(freq=D\)"
```

File: pandas/tests/indexes/period/test_partial_slicing.py [hunk 5]
```diff
@@ -59,6 +59,7 @@ def test_range_slice_day(self):
         didx = pd.date_range(start="2013/01/01", freq="D", periods=400)
         pidx = period_range(start="2013/01/01", freq="D", periods=400)
 
+        msg = f"slice indices must be integers or None or have an __index__ method"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_partial_slicing.py]:
```suggestion
        msg = "slice indices must be integers or None or have an __index__ method"
```

File: pandas/tests/indexes/period/test_partial_slicing.py [hunk 6]
```diff
@@ -81,13 +82,14 @@ def test_range_slice_day(self):
...
             for v in invalid:
-                with pytest.raises(TypeError):
+                with pytest.raises(TypeError, match=msg):
                     idx[v:]
 
     def test_range_slice_seconds(self):
         # GH#6716
         didx = pd.date_range(start="2013/01/01 09:00:00", freq="S", periods=4000)
         pidx = period_range(start="2013/01/01 09:00:00", freq="S", periods=4000)
+        msg = f"slice indices must be integers or None or have an __index__ method"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_partial_slicing.py]:
```suggestion
        msg = "slice indices must be integers or None or have an __index__ method"
```

File: pandas/tests/indexes/period/test_setops.py [hunk 7]
```diff
@@ -148,7 +148,8 @@ def test_union_misc(self, sort):
         # raise if different frequencies
         index = period_range("1/1/2000", "1/20/2000", freq="D")
         index2 = period_range("1/1/2000", "1/20/2000", freq="W-WED")
-        with pytest.raises(IncompatibleFrequency):
+        msg = fr"Input has different freq=W-WED from PeriodIndex\(freq=D\)"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_setops.py]:
```suggestion
        msg = r"Input has different freq=W-WED from PeriodIndex\(freq=D\)"
```

File: pandas/tests/indexes/period/test_setops.py [hunk 8]
```diff
@@ -180,11 +181,13 @@ def test_intersection(self, sort):
...
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_setops.py]:
```suggestion
        msg = r"Input has different freq=W-WED from PeriodIndex\(freq=D\)"
```

File: pandas/tests/indexes/period/test_setops.py [hunk 9]
```diff
@@ -180,11 +181,13 @@ def test_intersection(self, sort):
...
+        with pytest.raises(IncompatibleFrequency, match=msg):
             index.intersection(index2, sort=sort)
 
         index3 = period_range("1/1/2000", "1/20/2000", freq="2D")
-        with pytest.raises(IncompatibleFrequency):
+        msg = fr"Input has different freq=2D from PeriodIndex\(freq=D\)"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_setops.py]:
```suggestion
        msg = r"Input has different freq=2D from PeriodIndex\(freq=D\)"
```

**Review by ShaharNaveh — CHANGES_REQUESTED** [review]
In a few spots you are using f-strings where there isn't a reason to do so.

otherwise this is LGTM :)

**Comment by sumanau7** [comment]
@MomIsBestFriend Green.

File: pandas/tests/indexes/period/test_join.py [hunk 10]
```diff
@@ -39,5 +39,6 @@ def test_join_does_not_recur(self):
     def test_join_mismatched_freq_raises(self):
         index = period_range("1/1/2000", "1/20/2000", freq="D")
         index3 = period_range("1/1/2000", "1/20/2000", freq="2D")
-        with pytest.raises(IncompatibleFrequency):
+        msg = r".*Input has different freq=2D from PeriodIndex\(freq=D\)"
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/period/test_join.py]:
```suggestion
        msg = r"Input has different freq=2D from PeriodIndex\(freq=D\)"
```

**Review by ShaharNaveh — CHANGES_REQUESTED** [review]
Small nitpick

**Comment by jreback** [comment]
thanks @sumanau7

**Pull request merged**